flink-table/flink-sql-parser/pom.xml

```diff
@@ -38,25 +38,39 @@ under the License.
 	</properties>
 
 	<dependencies>
+		<dependency>
+			<groupId>com.google.guava</groupId>
+			<artifactId>guava</artifactId>
+			<version>${guava.version}</version>
+			<scope>provided</scope>
+		</dependency>
 		<dependency>
 			<groupId>org.apache.calcite</groupId>
 			<artifactId>calcite-core</artifactId>
 			<!-- When updating the Calcite version, make sure to update the dependency exclusions -->
 			<version>${calcite.version}</version>
 			<exclusions>
 				<!--
-				"mvn dependency:tree" as of Calcite 1.21:
+				"mvn dependency:tree" as of Calcite 1.22.0:
 
-				[INFO] +- org.apache.calcite:calcite-core:jar:1.21.0:compile
-				[INFO] |  +- org.apache.calcite.avatica:avatica-core:jar:1.15.0:compile
-				[INFO] |  \- com.google.guava:guava:jar:19.0:compile
+				[INFO] +- org.apache.calcite:calcite-core:jar:1.22.0:compile
+				[INFO] |  +- org.apache.calcite.avatica:avatica-core:jar:1.16.0:compile
+				[INFO] |  +- org.apiguardian:apiguardian-api:jar:1.1.0:compile
```

Comment:
Out of curiosity: what is `apiguardian` good for?

Reply:
Used to mark if some api is stable, the introduced version... Calcite uses that in 1.22 version in the code.

Reply:
It is a small github project though https://github.com/apiguardian-team/apiguardian

Reply:
This is unreasonable from my side, it only contains a single class `@API`. Why not add such class into Calcite code instead of introduce a dependency...

Reply:
We can remove it in Calcite 1.23, currently it does not affect Flink code though ~